logstash-core/lib/logstash/plugins/registry.rb

```diff
@@ -196,41 +197,53 @@ def lookup(type, plugin_name, &block)
     # a plugin and will do a lookup on the namespace of the required class to find a matching
     # plugin with the appropriate type.
     def legacy_lookup(type, plugin_name)
-      begin
-        path = "logstash/#{type}s/#{plugin_name}"
+      klass = load_plugin_class(type, plugin_name)
 
-        klass = begin
-          namespace_lookup(type, plugin_name)
-        rescue UnknownPlugin => e
-          # Plugin not registered. Try to load it.
-          begin
-            require path
-            namespace_lookup(type, plugin_name)
-          rescue LoadError => e
-            logger.error("Tried to load a plugin's code, but failed.", :exception => e, :path => path, :type => type, :name => plugin_name)
-            raise
-          end
-        end
+      has_alias = @alias_registry.alias?(type.to_java, plugin_name)
+      if !klass && has_alias
+        resolved_plugin_name = @alias_registry.original_from_alias(type.to_java, plugin_name)
+        logger.debug("Loading #{type} plugin #{resolved_plugin_name} via its alias #{plugin_name}...")
+        klass = load_plugin_class(type, resolved_plugin_name)
```

Comment:
We could obviate the need to hold `has_alias` by registering it here:

```suggestion
        klass = load_plugin_class(type, resolved_plugin_name)
        lazy_add(type, resolved_plugin_name, klass) if klass
```

Reply:
Suggestion applied :+1: